File: pset2/vigenere/vigenere.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>

int shift(char);

int main(int argc, string argv[])
{
// Check if there is only 1 key and if it is entirely alphabetical
    if(argc == 2)
    {
        for(int i=0, n = strlen(argv[1]); i < n; i++)
        {
            if(!(isalpha(argv[1][i])))
            {
                printf("Usage: ./vigenere keyword\n");
                return 1;
            }
        }
    }    
    else
    {
        printf("Usage: ./vigenere keyword\n");
        return 1;
    }
// Prompt user for plain text
    string p = get_string("plaintext: ");
    printf("ciphertext: ");
    int j = 0, k = strlen(argv[1]);
    for(int i = 0, n = strlen(p); i < n; i++)
    {
        if(islower(p[i]))
        {
            printf("%c", 'a'+ (p[i] + shift(argv[1][j % k]) - 'a') % 26);
            j++;
        }
        else if(isupper(p[i]))
        {
            printf("%c", 'A' + (p[i] + shift(argv[1][j % k]) - 'A') % 26);
            j++;
        }
        else
        {
            printf("%c", p[i]);
        }
    }
    printf("\n");
    return 0;
}

// Calculate the shift in letters using the key given
int shift(char c)
{
    int d;
    if(islower(c))
    {
        d = ((int) c - 'a');
    }
    else
    {
        d = ((int) c - 'A'); 
    }
    return d;
}

File: pset6/vigenere/vigenere.py
from sys import argv
from sys import exit
from cs50 import get_string

# Check validness of command line argument
if len(argv) != 2:
    print("usage: python vigenere.py keyword")
    exit(1)
if not argv[1].isalpha():
    print("keyword should be entirely alphabetical")
    exit(1)

# Generate keyword
keyword = []
argv[1] = argv[1].upper()
keylen = len(argv[1])
for i in range(keylen):
    keyword.append(int(ord(argv[1][i]) - ord('A')))

# Convert plain text to cipher text
raw = get_string("plaintext: ")
curkey = 0
print("ciphertext: ", end="")
for c in raw:
    if c.islower():
        print(f"{chr(ord('a') + (ord(c) + keyword[curkey] - ord('a')) % 26)}", end="")
        curkey = (curkey + 1) % keylen
    elif c.isupper():
        print(f"{chr(ord('A') + (ord(c) + keyword[curkey] - ord('A')) % 26)}", end="")
        curkey = (curkey + 1) % keylen
    else:
        print(f"{c}", end="")
print()